src/invoices/InvoiceDetails/ApproveConfirmationModal/useDuplicateInvoice.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import moment from 'moment';
+
+import {
+  batchFetch,
+  LIMIT_MAX,
+  DATE_FORMAT,
+  useShowCallout,
+} from '@folio/stripes-acq-components';
+
+const useDuplicateInvoice = (mutator, invoice) => {
+  const showCallout = useShowCallout();
+  const [invoices, setInvoices] = useState();
+  const formattedInvoiceDate = moment.utc(invoice.invoiceDate)?.format(DATE_FORMAT);
+
+  useEffect(() => {
+    mutator.duplicateInvoices.GET({
+      limit: `${LIMIT_MAX}`,
+      params: {
+        query: `id<>"${invoice.id}" AND
+          vendorInvoiceNo=="${invoice.vendorInvoiceNo}" AND
+          invoiceDate=="${formattedInvoiceDate}*" AND
+          vendorId=="${invoice.vendorId}"`,
+      },
+    })
+      .then(invoiceResp => {
+        setInvoices(invoiceResp);
+        const vendorIds = [...new Set(invoiceResp.map(inv => inv.vendorId))];
+
+        return batchFetch(mutator.vendors, vendorIds);
+      })
+      .then(vendorResp => (
+        setInvoices(prev => (
```

Comment:
and here?

Reply:
Firstly, I `setInvoices` after getting a response. The next set - it's update invoices with vendor info. So if the vendor request crashed user will see a duplicate invoice list without the vendor name.

Reply:
the thing is that invoices without vendors will be returned before vendors are arrived. And it might cause flickering of displaying list of duplicated invoices without and with vendors